### DISCUSSION TITLE
Can GCC linker (ar.exe) recognize MS VS generated library?

### DISCUSSION THREAD

**Discussion by bariserkus** [alandefreitas/matplotplusplus#108] [Q&A]
I was not able to link the matplot library in Windows using the mingw linker "ar.exe". I am using Codeblocks 20.03 and the mingw that comes with it, which I believe is quite new version. I believe it is Mingw, 64 bits, Build version 4.3.5 and gcc (or g++) version 8.1.0

I tried to run the example codes under Windows VS 2019 and was able to link the libraries and get the plots successfully.

I believe the binaries are generated with VS under Windows. If that is the case, is there a possibility that the matplot.lib files cannot be used with gcc? Anybody tried that?

Any suggestions?

**Comment by alandefreitas** [comment]
We don't really test and build the library with windows+mingw. 
Only MacOS + Clang, Linux + GCC, and windows + MSVC (CLion supports MSVC now). 
I'm not sure the binaries built with MSVC are adapted to work with MinGw.

You could try to build it from source, but there are still some problems. 

1) The conditionals in the build script usually go like:

```cmake
if (MSVC)
    # do this
else()
    # do that
endif()
```

and I'm not sure this differentiates properly between windows, windows + msvc, and windows + mingw.

2) There is so much code in this repository that MinGw might need one hack here and there.
